src/rgw/rgw_rados.h

```diff
@@ -2427,6 +2427,16 @@ class RGWRados
   const string& get_current_period_id() {
     return current_period.get_id();
   }
+
+  bool is_zonegroup_api_exist(const std::string& api) const {
+    if (!current_period.get_id().empty()) {
+      auto zonegroups_by_api = current_period.get_map().zonegroups_by_api;
```

Comment:
careful with `auto`, this will still make a copy. try `const auto&`

re: function name, maybe `has_zonegroup_api` or `does_zonegroup_api_exist`?